src/models/room.ts

```diff
@@ -2461,6 +2409,37 @@ export class Room extends TypedEventEmitter<EmittedEvents, RoomEventHandlerMap>
         return threadRoots;
     }
 
+    /**
+     * Add a receipt event to the room.
+     * @param {MatrixEvent} event The m.receipt event.
+     * @param {Boolean} synthetic True if this event is implicit.
+     */
+    public addReceipt(event: MatrixEvent, synthetic = false): void {
+        const content = event.getContent<ReceiptContent>();
+        Object.keys(content).forEach((eventId: string) => {
+            Object.keys(content[eventId]).forEach((receiptType: ReceiptType) => {
+                Object.keys(content[eventId][receiptType]).forEach((userId: string) => {
+                    const receipt = content[eventId][receiptType][userId] as any;
+                    const receiptForMainTimeline = !receipt.threafd_id || receipt.thread_id === MAIN_ROOM_TIMELINE;
+                    const receiptDestination = receiptForMainTimeline
+                        ? this
+                        : this.threads.get(receipt.thread_id);
```

Comment:
Could this be falsy?

Reply:
Looks like it can, and the strict null check CI is complaining about it too!